npm/init.js

```diff
@@ -16,15 +17,36 @@ function validateNpmVersion() {
     }
 }
 
+function redirectDetectDownload(starturl) {
+    if(process.env.https_proxy && process.env.https_proxy.length > 0) {
+        var mainurl = process.env.https_proxy + starturl.replace('https://', '/https/');
+        console.log(mainurl);
+        http.get(mainurl, function(res) {
+            if(res.statusCode == 302) {
```

Comment:
I suggest changing this if statement to handle both 301 and 302 status codes:
ֿֿ```if (res.statusCode == 301 || res.statusCode == 302) {```